test/integration/test/deck_test.go

```diff
@@ -379,137 +381,137 @@ func TestRerun(t *testing.T) {
+				return createdj.Before(&createdi)
 			})
+
 			if len(pjs.Items) > 0 {
 				if lastRun != nil && pjs.Items[0].CreationTimestamp.Before(lastRun) {
 					return false, nil
 				}
 				res = &pjs.Items[0]
 			}
+
 			return res != nil, nil
 		}); err != nil {
 			t.Fatalf("Failed waiting for job %q: %v", jobName, err)
 		}
 		return res
 	}
-	rerun := func(t *testing.T, jobName string, mode string) {
-		req, err := http.NewRequest("POST", fmt.Sprintf("http://localhost/rerun?mode=%v&prowjob=%v", mode, jobName), nil)
-		if err != nil {
-			t.Fatalf("Could not generate a request %v", err)
-		}
 
-		// Deck might not have been informed about the job config update, retry
-		// for this case.
-		waitDur := time.Second * 5
-		var lastErr error
-		for i := 0; i < 3; i++ {
-			lastErr = nil
-			res, err := http.DefaultClient.Do(req)
-			if err != nil {
-				lastErr = fmt.Errorf("could not make post request %v", err)
-				res.Body.Close()
-				break
-			}
-			// The only retry condition is status not ok
-			if res.StatusCode != http.StatusOK {
-				lastErr = fmt.Errorf("status not expected: %d", res.StatusCode)
-				res.Body.Close()
-				waitDur *= 2
-				time.Sleep(waitDur)
-				continue
-			}
-			body, err := io.ReadAll(res.Body)
-			if err != nil {
-				lastErr = fmt.Errorf("could not read body response %v", err)
-				res.Body.Close()
-				break
-			}
-			t.Logf("Response body: %s", string(body))
-			break
-		}
-		if lastErr != nil {
-			t.Fatalf("Failed trigger rerun: %v", lastErr)
-		}
-	}
-	jobToRerun := getLatestJob(t, jobName, nil)
-	rerunJobConfig = fmt.Sprintf(rerunJobConfigTemplate, jobName, "bar")
-	if err := updateJobConfig(context.Background(), kubeClient, rerunJobConfigFile, []byte(rerunJobConfig)); err != nil {
-		t.Fatalf("Failed update job config: %v", err)
-	}
-	var passed bool
-	// It may take some time for the new ProwJob to show up, so we will
-	// check every 30s interval three times for it to appear
-	latestRun := jobToRerun
-	for i := 0; i < 3; i++ {
-		time.Sleep(30 * time.Second)
-		rerun(t, jobToRerun.Name, "latest")
-		if latestRun = getLatestJob(t, jobName, &latestRun.CreationTimestamp); latestRun.Labels["foo"] == "bar" {
-			passed = true
-			break
-		}
-	}
-	if !passed {
-		t.Fatal("Expected updated job.")
+	// Wait for the first job to be created by horologium.
+	initialJob := getLatestJob(t, jobName, nil)
+
+	// Update the job configuration with a new label.
+	redeployJobConfig(fmt.Sprintf(rerunJobConfigTemplate, jobName, "bar"))
+
+	// Rerun the job using the latest config.
+	rerunJob(t, ctx, initialJob.Name, "latest")
+
+	// Wait until the desired ProwJob shows up.
+	latestJob := getLatestJob(t, jobName, &initialJob.CreationTimestamp)
+	if latestJob.Labels["foo"] != "bar" {
+		t.Fatalf("Failed waiting for ProwJob %q using latest config with foo=bar.", jobName)
 	}
 
+	// Prevent Deck from being too fast and recreating the new job in the same second
+	// as the previous one.
+	time.Sleep(1 * time.Second)
```

Comment:
What's the downside of having the second job created in the same second? Can we fix that instead of adding a sleep?

Reply:
Without the artifical delay, situations like this can happen:

```
pj[0] = name=3ee585ca-d93f-4fbc-9a09-b5c3baeeb2b6, created=2024-12-14 16:54:43 +0100 CET, id=1867961304296198144
pj[1] = name=b5578f85-ecb6-4741-b3d4-f9f33f099c59, created=2024-12-14 16:54:14 +0100 CET, id=1867961304296198144
pj[2] = name=094a01cf-782b-427a-b5e1-e02e551f604d, created=2024-12-14 16:54:43 +0100 CET, id=1867961425939402752
```

This leads to an instable sorting order, making the test flake. :-/